src/osd/PrimaryLogPG.cc

```diff
@@ -7067,6 +7067,37 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	  chunk_length += p.second.length;
 	}
 	if (chunk_length == obs.oi.size) {
+	  // set clone_size to 0
+	  hobject_t head_oid = soid.get_head();
+	  ObjectContextRef head_obc = get_object_context(head_oid, false);
+	  if (!head_obc) {
+	    dout(0) << "no snapset obc for " << head_oid << dendl;
+	    result = -ENOENT;
+	    break;
+	  }
+	  ctx->head_obc = head_obc;
+	  if (!ctx->lock_manager.take_write_lock(
+		head_oid,
+		head_obc)) {
+	    ceph_assert(0 == "head_obc can not take a lok");
+	  }
+
+	  head_obc->obs.oi.prior_version = head_obc->obs.oi.version;
+	  head_obc->obs.oi.version = ctx->at_version;
+	  SnapSet& snapset = head_obc->ssc->snapset;
+	  if (snapset.clone_size.count(soid.snap)) {
+	    snapset.clone_size[soid.snap] = 0;
```

Comment:
This isn't right, you can't adjust snapset.clone_size here.

In a larger sence, I think object_info_t::size and snapset::clone_size should reflect the logical size of the objects rather than the ondisk size.  Can we adjust scrub to check the manifest if the ondisk size doesn't match?